### ISSUE TITLE
FAIL : htpasswd does not exist

### ISSUE THREAD

**Issue by Lunik** [openshift/openshift-ansible#1960]
Before OpenShift installation, I generate an htpasswd file into `/root/.htpasswd`.
And I also set this variables into my hosts file:

```
openshift_master_identity_providers="[{'name': '{{ openshift_identity_provides.name }}', 'login': '{{ openshift_identity_provides.login }}', 'challenge': '{{ openshift_identity_provides.challenge }}', 'kind': '{{ openshift_identity_provides.kind }}', 'filename': '{{ openshift_identity_provides.filename }}'}]"
openshift_master_htpasswd_file="{{ openshift_htpasswd_file }}"
```

With:

```
ansible_ssh_user="root"

openshift_htpasswd_file: "/root/.htpasswd"

openshift_identity_provides:
  name: "htpasswd_auth"
  login: "true"
  challenge: "true"
  kind: "HTPasswdPasswordIdentityProvider"
  filename: "/etc/origin/master/htpasswd"
```

But it doesn't care ans send me this error:

```
TASK [openshift_master_facts : Set master facts] *******************************
fatal: [172.17.0.6]: FAILED! => {"failed": true, "msg": "the file_name '/root/.htpasswd' does not exist, or is not readable"}
```

**Comment by abutcher** [comment]
These settings are working for me where `openshift_master_htpasswd_file` is the local path on the system I'm running the playbooks from.

```
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_master_htpasswd_file="/tmp/htpasswd"
```

**Comment by detiber** [comment]
@abutcher this could be related to the delegate_to changes that you backed out.

**Comment by DanyC97** [comment]
@Lunik i suspect this issue is now solved in recent version. you mind closing it ? 

cc @vrutkovs candidate for closing as being stale issue

**Comment by vrutkovs** [comment]
This has been fixed in 3.9+ at least, please open a new issue if it still occurs
